api_app/api/routes/airlock.py

```diff
@@ -67,18 +75,123 @@ async def retrieve_airlock_request_by_id(airlock_request=Depends(get_airlock_req
+        user=Depends(get_current_airlock_manager_user),
+        workspace=Depends(get_deployed_workspace_by_id_from_path),
+        user_resource_repo=Depends(get_repository(UserResourceRepository)),
+        workspace_service_repo=Depends(get_repository(WorkspaceServiceRepository)),
+        operation_repo=Depends(get_repository(OperationRepository)),
+        airlock_request_repo=Depends(get_repository(AirlockRequestRepository)),
+        resource_template_repo=Depends(get_repository(ResourceTemplateRepository))) -> OperationInResponse:
+
+    if airlock_request.status != AirlockRequestStatus.InReview:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST,
+                            detail="Airlock request must be in 'in_review' status to create a Review User Resource")
+
+    try:
+        # Getting the right configuration
+        if airlock_request.requestType == AirlockRequestType.Import:
+            config = workspace.properties["airlock_review_config"]["import"]
+            workspace_id = config["workspace_id"]
+        else:
+            assert airlock_request.requestType == AirlockRequestType.Export
+            config = workspace.properties["airlock_review_config"]["export"]
+            workspace_id = workspace.id
+        workspace_service_id = config["workspace_service_id"]
+        user_resource_template_name = config["user_resource_template_name"]
+
+        logging.info(f"Going to create a user resource in {workspace_id} {workspace_service_id} {user_resource_template_name}")
+    except (KeyError, TypeError) as e:
+        logging.error(f"Failed to parse configuration: {e}")
+        raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                            detail=f"Failed to retrive Airlock Review configuration for workspace {workspace.id}.\
+                            Please ask your TRE administrator to check the configuration. Details: {str(e)}")
+
+    # Find workspace service to create user resource in
+    try:
+        workspace_service = workspace_service_repo.get_workspace_service_by_id(workspace_id=workspace_id, service_id=workspace_service_id)
+    except EntityDoesNotExist as e:
+        logging.error(f"Failed to get workspace serivce {workspace_service_id} for workspace {workspace_id}: {str(e)}")
+        raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                            detail=f"Failed to retrive Airlock Review configuration for workspace {workspace.id}.\
+                            Please ask your TRE administrator to check the configuration. Details: {str(e)}")
+
+    # Getting the SAS URL
+    airlock_request_sas_url = get_airlock_container_link(airlock_request, user, workspace)
+
+    # Now have all components for user resource, create an object for it
+    user_resource_create = UserResourceInCreate(
+        templateName=user_resource_template_name,
+        properties={
+            "display_name": "Review VM",
```

Comment:
Should we maybe say Airlock Review VM to be explicit? For cases where its an Export request they'll have a number of VMs showing up perhaps alongside regular VMs

Reply:
Yes good point, will rename. By the way, researchers will not see the Review VMs, as they only can see their own VMs. Same for Airlock Managers, they won't see the research VMs. But in any case it's best to be explicit!